lib/tests/message.test.ts

```diff
@@ -347,4 +355,610 @@ describe("Send message test", () => {
+    ]
+
+    const disconnect = channel.connect((message) => {
+      if (message.content.text !== undefined) {
+        messages.push(message.content.text)
+      }
+    })
+
+    for (const longMessage of longMessages) {
+      await channel.sendText(longMessage)
+      const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
+      await sleep(2000)
+    }
+
+    await waitForAllMessagesToBeDelivered(messages, longMessages)
+
+    for (const longMessage of longMessages) {
+      expect(messages).toContain(longMessage)
+    }
+
+    disconnect()
+  }, 30000)
+
+  test("should fail to send an empty or whitespace-only message", async () => {
+    const messages: string[] = []
+
+    const disconnect = channel.connect((message) => {
+      if (message.content.text !== undefined) {
+        messages.push(message.content.text)
+      }
+    })
+
+    try {
+      await channel.sendText("")
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toContain("Message text cannot be empty")
+    }
+
+    try {
+      await channel.sendText("   ")
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toContain("Message text cannot be empty")
+    }
+
+    expect(messages.length).toBe(0)
+
+    disconnect()
+  }, 30000)
+
+  test("should send and receive messages in various languages correctly", async () => {
+    const messages: string[] = []
+    const textMessages = [
+      "Hello",
+      "This is a test message",
+      "你好", // Chinese
+      "مرحبًا", // Arabic
+      "こんにちは", // Japanese
+      "안녕하세요", // Korean
+      "Hola", // Spanish
+    ]
+
+    const disconnect = channel.connect((message) => {
+      if (message.content.text !== undefined) {
+        messages.push(message.content.text)
+      }
+    })
+
+    for (const textMessage of textMessages) {
+      await channel.sendText(textMessage)
+      const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
+      await sleep(2000)
+    }
+
+    await waitForAllMessagesToBeDelivered(messages, textMessages)
+
+    for (const textMessage of textMessages) {
+      expect(messages).toContain(textMessage)
+    }
+
+    disconnect()
+  }, 30000)
+  //Task created CSK-284
+  test("should fail to edit a deleted message", async () => {
+    await channel.sendText("Test message")
+    await sleep(150) // history calls have around 130ms of cache time
+
+    const historyBeforeDelete = await channel.getHistory()
+    const messagesBeforeDelete: Message[] = historyBeforeDelete.messages
+    const sentMessage = messagesBeforeDelete[messagesBeforeDelete.length - 1]
+
+    await sentMessage.delete({ soft: false })
+    await sleep(150) // history calls have around 130ms of cache time
+
+    try {
+      await sentMessage.editText("Edited message")
+    } catch (error) {
+      expect(error.message).toContain("dfgdfgdfg not found")
+    }
+  }, 30000)
+
+  test("should delete toggle the message reaction", async () => {
```

Comment:
I don't understand this test

Reply:
I fixed description. Mainly logic is the same like in should toggle the message reaction test just checking undo reaction